mlrun/execution.py

```diff
@@ -893,6 +895,94 @@ def log_model(
+        upload: Optional[bool] = None,
+        **kwargs,
+    ) -> LLMPromptArtifact:
+        """Log an LLM prompt artifact and optionally upload it to the artifact store.
+
+        This function allows you to log a prompt artifact for large language model (LLM) usage. Prompts can be defined
+        as a string or by referencing a file path. Optionally, you can link the prompt to a parent model artifact and
+        provide metadata like a prompt legend (e.g., input variable mapping) and generation configuration.
+
+        Examples::
+
+            # Log an inline prompt
+            context.log_llm_prompt(
+                key="qa-prompt",
+                prompt_string="Q: {question}\nA:",
+                model_artifact="model:latest",
+                prompt_legend={"question": "user_input"},
+                generation_configuration={"temperature": 0.7, "max_tokens": 128},
+                tag="latest",
+            )
+
+            # Log from a file
+            project.log_llm_prompt(
+                key="summarization-prompt",
+                prompt_path="prompts/summarize.txt",
+                labels=["task=summarization"],
+            )
+
+        :param key: Unique name of the artifact.
+        :param prompt_string: Raw prompt text as a string. Cannot be used with `prompt_path`.
+        :param prompt_path: Path to a file containing the prompt content. Cannot be used with `prompt_string`.
+        :param prompt_legend: A dictionary that maps variables used in the prompt to their expected context keys.
+        :param model_artifact: Reference to the parent model (either `ModelArtifact` or model URI string).
+        :param generation_configuration: Dictionary of generation parameters (e.g., temperature, max_tokens).
+        :param description: Optional description of the prompt.
+        :param target_path: Path to write the artifact locally.
+        :param artifact_path: Path in the artifact store (defaults to project artifact path).
+        :param tag: Tag/version to assign to the prompt artifact.
+        :param labels: Labels to tag the artifact (e.g., list or dict of key-value pairs).
+        :param upload: Whether to upload the artifact to the store (defaults to True).
+        :param kwargs: Additional fields to pass to the `LLMPromptArtifact` constructor.
+
+        :returns: The logged `LLMPromptArtifact` object.
+        """
+
+        llm_prompt_spec = LLMPromptArtifactSpec(
+            prompt_string=prompt_string,
+            prompt_path=prompt_path,
+            prompt_legend=prompt_legend,
+            parent_uri=model_artifact.uri
+            if isinstance(model_artifact, ModelArtifact)
+            else model_artifact,
+            generation_configuration=generation_configuration,
+            target_path=target_path,
+            description=description,
+        )
+        llm_prompt = LLMPromptArtifact(
+            key=key,
+            spec=llm_prompt_spec,
+            **kwargs,
+        )
```

Comment:
If you implement `LLMPromptArtifact.__init__` then you don't need to do this step of generating the spec first, but rather just call that method.